### PR TITLE
Refactor: Rewrite `address` crate

### PR DESCRIPTION
**Pull request by Cai1Hsu**
This PR attempts to rewrite the existing address crate, currently named `address-v2`, which will be switched back to `address` during migration.

The existing implementation has the following issues:
- It uses interfaces instead of macros for code reuse, forcing any code using this library to import numerous unrelated interfaces. Rust-analyzer and AI struggle to detect them, frequently causing errors.
- Some type names are excessively verbose. While avoiding abbreviations is part of our coding style, fully qualified spelling for all words results in excessively long type names like `VirtualPageNumRange`. Rust-analyzer adds overly lengthy type hints, pushing code off-screen and significantly degrading the development experience.
- Usize conversions are cumbersome. This library uses `into` and `*` for conversions between usize and addresses.
- The const implementation is incomplete, with many operations not const-compatible.
- Coverage is insufficient; currently, all code in this crate achieves 100% coverage.

This PR currently:

- Added `PhysAddr` struct to represent physical addresses with arithmetic and formatting capabilities.
- Introduced `VaddrRange` struct for managing ranges of virtual addresses, including methods for creation, merging, and intersection.
- Implemented `VirtAddr` struct for virtual addresses, supporting pointer conversions and arithmetic operations.
- Created macros for implementing address and range types to reduce boilerplate code.
- Added comprehensive unit tests for all new functionalities, covering creation, arithmetic, alignment, and range operations.

TODO:
- [x] The type representing Pages and its corresponding Range type are not yet implemented. We may consider redesigning such types to support multiple page sizes.
- [ ] Migrate the existing code to use this crate(may not be in this PR).

## Summary by CodeRabbit

* **New Features**
  * Added address-v2 library: strongly-typed Phys/Virt address types, ranges, pages, and page-ranges with constructors, alignment helpers, arithmetic/bitwise ops, pointer/reference conversions, and iterators; virt and phys namespaces expose AddrIterator and PageIterator aliases (misspelling corrected).
* **Chores**
  * Added the new library to the workspace manifest.
* **Tests**
  * Comprehensive unit tests covering APIs, edge cases, alignment/page behavior, conversions, and iteration (no_std-compatible).

### CODE REVIEW

File: libraries/address-v2/src/addr_range_base.rs [hunk 1]
```diff
@@ -0,0 +1,529 @@
...
+            }
+
+            #[inline]
+            pub const fn new_unchecked(range: $range_type, step: usize) -> Self {
+                Self {
+                    current: range.start,
+                    end: range.end,
+                    step,
+                }
+            }
```
[the remark this answers is not in the document]
Comment (Cai1Hsu) [reply libraries/address-v2/src/addr_range_base.rs]: this is necessary in theory, but feels not *unchecked*

**Comment by Cai1Hsu** [comment]
@coderabbitai review pls
